modules/hal_silabs/wiseconnect/CMakeLists.txt

```diff
@@ -151,5 +152,32 @@ if(CONFIG_WISECONNECT_NETWORK_STACK)
   )
 endif() # CONFIG_WISECONNECT_NETWORK_STACK
 
+# Sleeptimer for 917 devices
+if(CONFIG_SOC_SILABS_SLEEPTIMER)
+  zephyr_include_directories(
+    ${SISDK_DIR}/platform/service/sleeptimer/inc
+    ${SISDK_DIR}/platform/service/sleeptimer/src
+    ${SISDK_DIR}/platform/service/sleeptimer/config
+  )
+
+  zephyr_library_sources(
+    ${SISDK_DIR}/platform/service/sleeptimer/src/sl_sleeptimer.c
+    ${WISECONNECT_DIR}/components/device/silabs/si91x/mcu/drivers/service/sleeptimer/src/sl_sleeptimer_hal_si91x_sysrtc.c
+  )
+  zephyr_code_relocate(FILES
+    ${ZEPHYR_BASE}/arch/arm/core/cortex_m/isr_wrapper.c
+    ${SISDK_DIR}/platform/service/sleeptimer/src/sl_sleeptimer.c
+    ${WISECONNECT_DIR}/components/device/silabs/si91x/mcu/drivers/service/sleeptimer/src/sl_sleeptimer_hal_si91x_sysrtc.c
+    ${WISECONNECT_DIR}/components/device/silabs/si91x/mcu/drivers/peripheral_drivers/src/rsi_sysrtc.c
+    LOCATION RAM
+    FILTER ".*"
+  )
+  zephyr_compile_definitions(
+    SL_CATALOG_SLEEPTIMER_PRESENT
+    SL_CODE_COMPONENT_SLEEPTIMER=sleeptimer
+    SL_CODE_COMPONENT_HAL_SYSRTC=hal_sysrtc
+  )
+endif() # CONFIG_SOC_SILABS_SLEEPTIMER
```

Comment:
This section is redundant with `modules/hal_silabs/simplicity_sdk/CMakeLists.txt`. Maybe we should split the `SOC_SILABS_SLEEPTIMER` symbol in smaller parts:

```
config SOC_SILABS_SLEEPTIMER
        depends on SOC_SILABS_SIWX91X_SYSRTC || SOC_SILABS_S2_SYSRTC
```

@asmellby what is your opinion here?

Reply:
We need to clean up the HAL level Kconfig symbols in general. Symbols are being mixed between the GSDK and SiSDK HALs, and the scope of symbols is often tied to their _use_ in Zephyr, rather than the _feature_ they provide from the HAL. We should redo all the SiSDK symbols to properly map to SiSDK drivers, which will make them more easily composable and reusable for SiWx91x. But that should be a separate PR from this one.